finder-app/writer.c

```diff
@@ -0,0 +1,59 @@
+
+/*
+ *  
+ *  Author     : Vignesh Vadivel
+ *  email      : viva9969@colorado.edu 
+ *  Course     : Advanced Embedded Software Development
+ *  Assignment : 02
+ * 
+ */
+
+/***************************************** HEADER_FILES *******************************************************/
+#include <syslog.h>
+#include <fcntl.h>
+#include <stddef.h>
+#include <unistd.h>
+#include <string.h>
+
+/***************************************** Main Function *******************************************************/
+int main(int argc, char* argv[]){
+  if (argc!=3){                                                                  // Check valid number of arguments //
+    openlog(NULL, 0, LOG_USER);                                                  // Open Log file //
+    syslog(LOG_ERR, "Invalid Number of Arguments: %d\n", argc);                  // Log the error message //
+    closelog();                                                                  // Close log file //
+    return 1;
+  }
+  
+  int fp = open(argv[1], O_CREAT | O_WRONLY , S_IRWXU | S_IRWXG | S_IRWXO);      // Open the file to write //
+  if (fp != -1){                                                                 // Check if the file pointer is valid //
+    int i = strlen(argv[2]);                                                     // Calculate the total characters to be written into the file //
```

Comment:
@Vignesh-Vadivel suggest using a variable name rather than using argv references everywhere.

Reply:
argv basically have command line arguments in it. So maybe I can store the content in some other local variables. For instance, in this case, argv[1] will have array of characters. So may be I can store in it a char pointer, and I can access it. In that way it would be easy to understand the code.